makefiles/features_check.inc.mk

```diff
@@ -1,20 +1,28 @@
 # Check if all required FEATURES are provided
 
+# Features that are used without taking "one out of" dependencies into account
+FEATURES_USED_SO_FAR := $(sort $(FEATURES_REQUIRED) $(FEATURES_OPTIONAL_USED))
```

Comment:
Why did you move this one? `FEATURES_OPTIONAL_USED` is defined below no?

Reply:
In line 7 this is accessed and needs to be provided by then. `FEATURES_OPTIONAL_USED` on the other hand depends on the usable features, which depends on this.

Reply:
Ah so it catches the previous iteration?